### ISSUE TITLE
Problem when having multiple OneToOne relationships

### ISSUE THREAD

**Issue by HenryAngulo** [mikro-orm/mikro-orm#2845]
**Describe the bug**
I have an entity (User) that is the inverse side of two OneToOne relationship and i can't populate both at the same time. Besides, if i don't try to populate at all (do a find without options), the resulting query always do a join with the first relationship.

**To Reproduce**
1. The entities involved are:
```
@Entity()
export class User extends CustomBaseEntity {
  @PrimaryKey({ type: `uuid`, defaultRaw: `uuid_generate_v4()` })
  id!: string
  @Unique()
  email: string
  @Property()
  password!: string
  ...other properties
  //relationships
  @ManyToOne({ entity: () => Role })
  role!: Role
  @OneToOne(() => Professional, (professional) => professional.user)
  professional?: Professional
  @OneToOne(() => Admissionist, (admissionist) => admissionist.user)
  admissionist?: Admissionist
  ...

}
```
```
@Entity()
export class Admissionist extends CustomBaseEntity {
  @PrimaryKey({ type: `uuid`, defaultRaw: `uuid_generate_v4()` })
  id!: string
  ...other properties
  //relationships
  @OneToOne(() => User, (user) => user.admissionist, { owner: true })
  user!: User
  ...
}
```
```
@Entity()
export class Professional extends CustomBaseEntity {
  @PrimaryKey({ type: `uuid`, defaultRaw: `uuid_generate_v4()` })
  id!: string
  ...other properties
  //relationships
  @OneToOne(() => User, (user) => user.professional, {
    owner: true,
  })
  user!: User
...
```
2. If i try to find user without any populate, it try to populate the id of the first relationship
```
    const em = await getEntityManager()
    const users = await em.find(User, {})
```
The resulting query and result are the following

```
//Query
select "u0".*, "p1"."id" as "professionalId" from "Users" as "u0" left join "Professionals" as "p1" on "u0"."id" = "p1"."userId"
//Result
 [
        {
            "id": "3a0f3f5a-8a60-414f-a462-7fce455044bc",
            ...
            "professional": "4e5c0708-f168-46ef-8ba3-8ba1d6a393fe"
        },
        {
            "id": "e1636b7f-c620-4f38-aad1-ecc1c08e47d0",
             ... 
            //this user has no related professional
            "professional": null
        }
    ]
```
3. If i try to find using populate with one of the relationship it works correctly (it also works if the populate is only 'professional')
```
    const em = await getEntityManager()
    const users = await em.find(User, {}, { populate: [`admissionist`] })
```
The resulting queries and result are the following

```
//Queries
select "u0".*, "a1"."id" as "admissionistId" from "Users" as "u0" left join "Admissionists" as "a1" on "u0"."id" = "a1"."userId"
select "a0".* from "Admissionists" as "a0" where "a0"."id" in ('5bcff1e9-98d8-4d27-99ca-a032d85ce67b') order by "a0"."id" asc
//Result
 [
        {
            "id": "3a0f3f5a-8a60-414f-a462-7fce455044bc",
            ...
            "admissionist": {... populated admissionist}
        },
        {
            "id": "e1636b7f-c620-4f38-aad1-ecc1c08e47d0",
             ...
             //this user has no related admissionist
            "admissionist": null
        }
    ]
```
4. If i try to populate both relationships, it only populate one.
```
    const em = await getEntityManager()
    const users = await em.find(
      User,
      {},
      { populate: [`admissionist`, `professional`] },
    )
```
The resulting queries and result are the following

```
//Queries
select "u0".*, "a1"."id" as "admissionistId" from "Users" as "u0" left join "Admissionists" as "a1" on "u0"."id" = "a1"."userId"
select "a0".* from "Admissionists" as "a0" where "a0"."id" in ('5bcff1e9-98d8-4d27-99ca-a032d85ce67b') order by "a0"."id" asc
//Result
 [
        {
            "id": "3a0f3f5a-8a60-414f-a462-7fce455044bc",
            ...
            "admissionist": {... populated admissionist}
        },
        {
            "id": "e1636b7f-c620-4f38-aad1-ecc1c08e47d0",
             ...
             //this user has no related admissionist
            "admissionist": null
        }
    ]
```
**Expected behavior**
I don't know if there is something obvious i'm missing but i would expect:
1. If i don't explicitly try to populate relationships that there wouldn't be any joins
2. I could populate as much relationships as i want at the same time and not only one at a time.

Thanks in advance and for the great work!
**Versions**

| Dependency | Version |
| - | - |
| node | 14.17.0 |
| typescript | 4.5.5 |
| mikro-orm | 5.0.4 |
| your-driver | "@mikro-orm/postgresql": "^5.0.4" |

**Comment by B4nan** [comment]
Please always upgrade to latest before filing issues. This has been fixed in 5.0.5.

**Comment by HenryAngulo** [comment]
You're right, sorry, next time i'll check that first. I've only started using it last week and i guess i'm not used to projects as active as yours.  Thanks!
